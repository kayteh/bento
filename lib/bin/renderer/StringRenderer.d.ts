export default class StringRenderer {
    _data: string;
    r: (str: string) => void;
    data: () => string;
}
