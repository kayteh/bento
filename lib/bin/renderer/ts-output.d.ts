import pbjs from 'protobufjs';
export declare const translateProtoPath: (filePath: string) => string;
export declare const postWriteTasks: (filePath: string) => void;
export declare const transformRpcName: (inp: string) => string;
export declare const resolveJSType: (t: string) => string;
export declare const renderType: (t: pbjs.Type) => string;
export declare const renderService: (s: pbjs.Service) => string;
