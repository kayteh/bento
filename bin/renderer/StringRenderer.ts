export default class StringRenderer {
  _data: string = ''

  r = (str: string) => {
    this._data += `${str}\n`
  }

  data = () => {
    return this._data
  }
}
