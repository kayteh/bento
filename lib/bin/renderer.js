#!/usr/bin/env node
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const protobufjs_1 = __importDefault(require("protobufjs"));
const path_1 = __importDefault(require("path"));
const fs_extra_1 = __importDefault(require("fs-extra"));
const glob_1 = __importDefault(require("glob"));
const tsOutput = __importStar(require("./renderer/ts-output"));
const StringRenderer_1 = __importDefault(require("./renderer/StringRenderer"));
const BENTO_LOC = process.env.BENTO_LOC || '@kayteh/bento';
const flatten = (array) => array.reduce((acc, cur) => [...acc, ...cur], []);
exports.writeOut = (filepath, content) => __awaiter(this, void 0, void 0, function* () {
    // still assuming ts.
    const fn = filepath.replace('.proto', '.bento.ts');
    const base = path_1.default.basename(fn);
    const atom = fn.replace(base, `.${base}~`);
    yield fs_extra_1.default.writeFile(atom, content, { encoding: 'utf8' });
    yield fs_extra_1.default.rename(atom, fn);
});
exports.render = (rd) => {
    // assuming ts renderer for now
    // types first
    const { r, data } = new StringRenderer_1.default();
    r(`/**
 * GENERATED FILE. This file was generated by @kayteh/bento. Editing it is a bad idea.
 * @generated
 */`);
    r(`import Bento, { IBentoTransport } from '${BENTO_LOC}'`);
    for (let t of rd.types) {
        r(tsOutput.renderType(t));
    }
    for (let s of rd.services) {
        r(tsOutput.renderService(s));
    }
    return data();
};
exports.prepRender = (filePath, root) => {
    const rd = {
        filePath,
        types: [],
        services: []
    };
    console.log({ root });
    for (const obj of root.nestedArray) {
        // if (!filePath.includes(obj.filename || '')) {
        //   continue
        // }
        if (obj instanceof protobufjs_1.default.Type) {
            // console.log('got type', obj.name)
            if (obj.options !== undefined && obj.options['render.exclude'] === true) {
                continue;
            }
            rd.types.push(obj);
            continue;
        }
        if (obj instanceof protobufjs_1.default.Service) {
            // console.log('got service', obj.name)
            rd.services.push(obj);
            continue;
        }
    }
    return rd;
};
const injectResolvePath = (root, paths) => {
    const mainFiles = [];
    // console.log(paths)
    root.resolvePath = (origin, target) => {
        // console.log({ origin, target })
        let normOrigin = protobufjs_1.default.util.path.normalize(origin);
        let normTarget = protobufjs_1.default.util.path.normalize(target);
        if (!normOrigin) {
            mainFiles.push(normTarget);
        }
        // console.log({ normOrigin, normTarget })
        let resolved = protobufjs_1.default.util.path.resolve(normOrigin, normTarget, true);
        let idx = resolved.lastIndexOf('google/protobuf/');
        if (idx > -1) {
            let altname = resolved.substring(idx);
            // console.log({ altname, resolved })
            if (altname in protobufjs_1.default.common) {
                resolved = altname;
            }
        }
        // console.log({ resolved })
        // console.log(1)
        if (fs_extra_1.default.existsSync(resolved)) {
            // console.log(2)
            return resolved;
        }
        for (let path of paths) {
            let iresolved = protobufjs_1.default.util.path.resolve(path + '/', target);
            // console.log(3, { path, target })
            if (fs_extra_1.default.existsSync(iresolved)) {
                // console.log(4, { iresolved })
                return iresolved;
            }
        }
        // console.log(5, { resolved })
        return resolved;
    };
};
const processFile = (fileName) => __awaiter(this, void 0, void 0, function* () {
    if (fileName.includes('node_modules')) {
        return;
    }
    const root = new protobufjs_1.default.Root();
    const paths = [
        path_1.default.relative(process.cwd(), path_1.default.dirname(require.resolve('protobufjs'))),
        path_1.default.resolve(path_1.default.join(__dirname, '../../proto-inc'))
    ];
    injectResolvePath(root, paths);
    const f = yield root.load(fileName);
    const rd = exports.prepRender(fileName, f);
    const rendered = exports.render(rd);
    yield exports.writeOut(fileName, rendered);
    tsOutput.postWriteTasks(fileName);
    // TODO: add chalk
    console.log(`📝 Wrote out ${fileName}`);
    console.log(`-- Found RPCs: ${rd.services.map(v => v.name).join(', ')}`);
    console.log(`-- Found types: ${rd.types.map(v => v.name).join(', ')}`);
    console.log('');
});
exports.run = (globPaths) => __awaiter(this, void 0, void 0, function* () {
    const globOpts = {
        ignore: ['./node_modules/**/*.proto', path_1.default.join(__dirname, 'proto-inc/**/*.proto')]
    };
    // console.log({ globPaths })
    const globs = globPaths.map(p => glob_1.default.sync(p, globOpts));
    const files = flatten(globs);
    // console.log(files)
    if (files.length === 0) {
        console.error('No files found.');
    }
    for (const file of files) {
        yield processFile(file);
    }
});
if (require.main === module) {
    exports.run(process.argv.slice(2)).catch((e) => {
        console.error(e);
    });
}
