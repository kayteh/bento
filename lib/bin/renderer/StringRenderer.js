"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
class StringRenderer {
    constructor() {
        this._data = '';
        this.r = (str) => {
            this._data += `${str}\n`;
        };
        this.data = () => {
            return this._data;
        };
    }
}
exports.default = StringRenderer;
