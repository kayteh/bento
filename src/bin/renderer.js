#!/usr/bin/env node
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
var _this = this;
exports.__esModule = true;
var protobufjs_1 = require("protobufjs");
var path_1 = require("path");
var fs_extra_1 = require("fs-extra");
var glob_1 = require("glob");
var tsOutput = require("./renderer/ts-output");
var StringRenderer_1 = require("./renderer/StringRenderer");
var BENTO_LOC = process.env.BENTO_LOC || '@kayteh/bento';
var flatten = function (array) { return array.reduce(function (acc, cur) { return acc.concat(cur); }, []); };
exports.writeOut = function (filepath, content) { return __awaiter(_this, void 0, void 0, function () {
    var fn, base, atom;
    return __generator(this, function (_a) {
        switch (_a.label) {
            case 0:
                fn = filepath.replace('.proto', '.bento.ts');
                base = path_1["default"].basename(fn);
                atom = fn.replace(base, "." + base + "~");
                return [4 /*yield*/, fs_extra_1["default"].writeFile(atom, content, { encoding: 'utf8' })];
            case 1:
                _a.sent();
                return [4 /*yield*/, fs_extra_1["default"].rename(atom, fn)];
            case 2:
                _a.sent();
                return [2 /*return*/];
        }
    });
}); };
exports.render = function (rd) {
    // assuming ts renderer for now
    // types first
    var _a = new StringRenderer_1["default"](), r = _a.r, data = _a.data;
    r("/**\n * GENERATED FILE. This file was generated by @kayteh/bento. Editing it is a bad idea.\n * @generated\n */");
    r("import Bento, { IBentoTransport } from '" + BENTO_LOC + "'");
    for (var _i = 0, _b = rd.types; _i < _b.length; _i++) {
        var t = _b[_i];
        r(tsOutput.renderType(t));
    }
    for (var _c = 0, _d = rd.services; _c < _d.length; _c++) {
        var s = _d[_c];
        r(tsOutput.renderService(s));
    }
    return data();
};
exports.prepRender = function (filePath, root) {
    var rd = {
        filePath: filePath,
        types: [],
        services: []
    };
    console.log({ root: root });
    for (var _i = 0, _a = root.nestedArray; _i < _a.length; _i++) {
        var obj = _a[_i];
        // if (!filePath.includes(obj.filename || '')) {
        //   continue
        // }
        if (obj instanceof protobufjs_1["default"].Type) {
            // console.log('got type', obj.name)
            if (obj.options !== undefined && obj.options['render.exclude'] === true) {
                continue;
            }
            rd.types.push(obj);
            continue;
        }
        if (obj instanceof protobufjs_1["default"].Service) {
            // console.log('got service', obj.name)
            rd.services.push(obj);
            continue;
        }
    }
    return rd;
};
var injectResolvePath = function (root, paths) {
    var mainFiles = [];
    // console.log(paths)
    root.resolvePath = function (origin, target) {
        // console.log({ origin, target })
        var normOrigin = protobufjs_1["default"].util.path.normalize(origin);
        var normTarget = protobufjs_1["default"].util.path.normalize(target);
        if (!normOrigin) {
            mainFiles.push(normTarget);
        }
        // console.log({ normOrigin, normTarget })
        var resolved = protobufjs_1["default"].util.path.resolve(normOrigin, normTarget, true);
        var idx = resolved.lastIndexOf('google/protobuf/');
        if (idx > -1) {
            var altname = resolved.substring(idx);
            // console.log({ altname, resolved })
            if (altname in protobufjs_1["default"].common) {
                resolved = altname;
            }
        }
        // console.log({ resolved })
        // console.log(1)
        if (fs_extra_1["default"].existsSync(resolved)) {
            // console.log(2)
            return resolved;
        }
        for (var _i = 0, paths_1 = paths; _i < paths_1.length; _i++) {
            var path_2 = paths_1[_i];
            var iresolved = protobufjs_1["default"].util.path.resolve(path_2 + '/', target);
            // console.log(3, { path, target })
            if (fs_extra_1["default"].existsSync(iresolved)) {
                // console.log(4, { iresolved })
                return iresolved;
            }
        }
        // console.log(5, { resolved })
        return resolved;
    };
};
var processFile = function (fileName) { return __awaiter(_this, void 0, void 0, function () {
    var root, paths, f, rd, rendered;
    return __generator(this, function (_a) {
        switch (_a.label) {
            case 0:
                if (fileName.includes('node_modules')) {
                    return [2 /*return*/];
                }
                root = new protobufjs_1["default"].Root();
                paths = [
                    path_1["default"].relative(process.cwd(), path_1["default"].dirname(require.resolve('protobufjs'))),
                    path_1["default"].resolve(path_1["default"].join(__dirname, '../../proto-inc'))
                ];
                injectResolvePath(root, paths);
                return [4 /*yield*/, root.load(fileName)];
            case 1:
                f = _a.sent();
                rd = exports.prepRender(fileName, f);
                rendered = exports.render(rd);
                return [4 /*yield*/, exports.writeOut(fileName, rendered)];
            case 2:
                _a.sent();
                tsOutput.postWriteTasks(fileName);
                // TODO: add chalk
                console.log("\uD83D\uDCDD Wrote out " + fileName);
                console.log("-- Found RPCs: " + rd.services.map(function (v) { return v.name; }).join(', '));
                console.log("-- Found types: " + rd.types.map(function (v) { return v.name; }).join(', '));
                console.log('');
                return [2 /*return*/];
        }
    });
}); };
exports.run = function (globPaths) { return __awaiter(_this, void 0, void 0, function () {
    var globOpts, globs, files, _i, files_1, file;
    return __generator(this, function (_a) {
        switch (_a.label) {
            case 0:
                globOpts = {
                    ignore: ['./node_modules/**/*.proto', path_1["default"].join(__dirname, 'proto-inc/**/*.proto')]
                };
                globs = globPaths.map(function (p) { return glob_1["default"].sync(p, globOpts); });
                files = flatten(globs);
                // console.log(files)
                if (files.length === 0) {
                    console.error('No files found.');
                }
                _i = 0, files_1 = files;
                _a.label = 1;
            case 1:
                if (!(_i < files_1.length)) return [3 /*break*/, 4];
                file = files_1[_i];
                return [4 /*yield*/, processFile(file)];
            case 2:
                _a.sent();
                _a.label = 3;
            case 3:
                _i++;
                return [3 /*break*/, 1];
            case 4: return [2 /*return*/];
        }
    });
}); };
if (require.main === module) {
    exports.run(process.argv.slice(2))["catch"](function (e) {
        console.error(e);
    });
}
