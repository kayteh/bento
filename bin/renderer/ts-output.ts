import pbjs from 'protobufjs'
import StringRenderer from './StringRenderer'
import * as cp from 'child_process'
import camelCase from 'camel-case'

export const translateProtoPath = (filePath: string) => filePath.replace('.proto', '.bento.ts')
export const postWriteTasks = (filePath: string) => {
  // try tslint
  cp.execSync(`yarn tslint ${translateProtoPath(filePath)} --fix`, { encoding: 'utf8', shell: 'sh', stdio: 'ignore' })
}

export const transformRpcName = (inp: string) => camelCase(inp)

export const resolveJSType = (t: string) => {
  switch (t) {
    case 'bool': return 'boolean'
    case 'float': return 'number'
  }

  return t
}
export const renderType = (t: pbjs.Type): string => {
  const { r, data } = new StringRenderer()

  r(`export type ${t.name} = {`)
  for (let field of t.fieldsArray) {
    r(`  ${field.name}${field.required ? '' : '?'}: ${resolveJSType(field.type)}${field.repeated ? '[]' : ''}`)
  }
  r(`}`)

  return data()
}

export const renderService = (s: pbjs.Service): string => {
  const { r, data } = new StringRenderer()

  // service interface
  r(`export interface I${s.name}Service {`)
  for (let rpc of s.methodsArray) {
    r(`  ${transformRpcName(rpc.name)} (ctx: any, request: ${rpc.requestType}): Promise<${rpc.responseType}> | ${rpc.responseType}`)
  }
  r(`}`)

  // client class
  r(`export class ${s.name}Client {`)
  r(`  static __SERVICE__: string = '${s.name}'`)
  r(`  constructor(private bento: Bento, private transport?: IBentoTransport) {}`)
  for (let rpc of s.methodsArray) {
    r(`  async ${transformRpcName(rpc.name)} (request: ${rpc.requestType}): Promise<${rpc.responseType}> {`)
    r(`    return this.bento.makeRequest(this.transport || undefined, '${s.name}', '${rpc.name}', request)`)
    r(`  }`)
  }
  r(`}`)

  return data()
}
