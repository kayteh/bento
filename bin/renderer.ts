#!/usr/bin/env node

import pbjs from 'protobufjs'
import path from 'path'
import fs from 'fs-extra'
import glob from 'glob'

import * as tsOutput from './renderer/ts-output'
import StringRenderer from './renderer/StringRenderer'

const BENTO_LOC = process.env.BENTO_LOC || '@kayteh/bento'

// console.log({ cwd: process.cwd() })

type RenderData = {
  filePath: string
  types: pbjs.Type[]
  services: pbjs.Service[]
}

const flatten = <T> (array: T[][]) => array.reduce((acc, cur) => [...acc, ...cur], [])

export const writeOut = async (filepath: string, content: string) => {
  // still assuming ts.
  const fn = filepath.replace('.proto', '.bento.ts')
  const base = path.basename(fn)
  const atom = fn.replace(base, `.${base}~`)
  await fs.writeFile(atom, content, { encoding: 'utf8' })
  await fs.rename(atom, fn)
}

export const render = (rd: RenderData): string => {
  // assuming ts renderer for now
  // types first
  const { r, data } = new StringRenderer()
  r(`/**
 * GENERATED FILE. This file was generated by @kayteh/bento. Editing it is a bad idea.
 * @generated
 */`)
  r(`import Bento, { IBentoTransport } from '${BENTO_LOC}'`)

  for (let t of rd.types) {
    r(tsOutput.renderType(t))
  }

  for (let s of rd.services) {
    r(tsOutput.renderService(s))
  }

  return data()
}

export const prepRender = (filePath: string, root: pbjs.Root): RenderData => {
  const rd: RenderData = {
    filePath,
    types: [],
    services: []
  }
  for (const obj of root.nestedArray) {
    if (!filePath.includes(obj.filename || '')) {
      continue
    }

    if (obj instanceof pbjs.Type) {
      // console.log('got type', obj.name)
      if (obj.options !== undefined && obj.options['render.exclude'] === true) {
        continue
      }

      rd.types.push(obj)
      continue
    }

    if (obj instanceof pbjs.Service) {
      // console.log('got service', obj.name)
      rd.services.push(obj)
      continue
    }
  }
  return rd
}

const injectResolvePath = (root: pbjs.Root, paths: string[]) => {
  const mainFiles = []
  // console.log(paths)
  root.resolvePath = (origin, target) => {
    let normOrigin = pbjs.util.path.normalize(origin)
    let normTarget = pbjs.util.path.normalize(target)
    if (!normOrigin) {
      mainFiles.push(normTarget)
    }

    // console.log({ normOrigin, normTarget })

    let resolved = pbjs.util.path.resolve(normOrigin, normTarget, true)
    let idx = resolved.lastIndexOf('google/protobuf/')
    if (idx > -1) {
      let altname = resolved.substring(idx)
      // console.log({ altname, resolved })
      if (altname in pbjs.common) {
        resolved = altname
      }
    }

    // console.log({ resolved })

    // console.log(1)
    if (fs.existsSync(resolved)) {
      // console.log(2)
      return resolved
    }

    for (let path of paths) {
      let iresolved = pbjs.util.path.resolve(path + '/', target)
      // console.log(3, { path, target })
      if (fs.existsSync(iresolved)) {
        // console.log(4, { iresolved })
        return iresolved
      }
    }

    // console.log(5, { resolved })

    return resolved
  }
}

const processFile = async (fileName: string) => {
  if (fileName.includes('node_modules')) {
    return
  }

  const root = new pbjs.Root()
  const paths = [
    path.relative(process.cwd(), path.dirname(require.resolve('protobufjs'))), // pbjs common
    path.resolve(path.join(__dirname, '../../proto-inc'))
  ]
  injectResolvePath(root, paths)
  const f = await root.load(fileName)

  const rd: RenderData = prepRender(fileName, f)
  const rendered = render(rd)

  await writeOut(fileName, rendered)

  tsOutput.postWriteTasks(fileName)

  // TODO: add chalk
  console.log(`📝 Wrote out ${fileName}`)
  console.log(`-- Found RPCs: ${rd.services.map(v => v.name).join(', ')}`)
  console.log(`-- Found types: ${rd.types.map(v => v.name).join(', ')}`)
  console.log('')
}

export const run = async (globPaths: string[]) => {
  const globOpts = {
    ignore: ['./node_modules/**/*.proto', path.join(__dirname, 'proto-inc/**/*.proto')]
  }

  // console.log({ globPaths })

  const globs = globPaths.map(p => glob.sync(p, globOpts))

  const files = flatten(globs)
  // console.log(files)

  if (files.length === 0) {
    console.error('No files found.')
  }

  for (const file of files) {
    await processFile(file)
  }
}

if (require.main === module) {
  run(process.argv.slice(2)).catch((e) => {
    console.error(e)
  })
}
