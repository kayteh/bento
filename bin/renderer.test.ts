import { run } from './renderer'
describe('Renderer', () => {
  it('runs', async () => {
    await run('./**/*.proto')
  })
})
