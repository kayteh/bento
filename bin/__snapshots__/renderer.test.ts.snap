// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderer renders a snapshot 1`] = `
"/**
 * GENERATED FILE. This file was generated by @kayteh/bento. Editing it is a bad idea.
 * @generated
 */
import Bento, { IBentoTransport } from '@kayteh/bento'
export type HelloWorld = {
  username?: string
  help: string
  everyone?: string[]
}

export interface IHelloRpcService {
  hello (ctx: any, request: HelloWorld): Promise<HelloWorld> | HelloWorld
}
export class HelloRpcClient {
  static __SERVICE__: string = 'HelloRpc'
  constructor(private bento: Bento, private transport?: IBentoTransport) {}
  async hello (request: HelloWorld): Promise<HelloWorld> {
    return this.bento.makeRequest(this.transport || undefined, 'HelloRpc', 'Hello', request)
  }
}

"
`;
